app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -211,6 +247,161 @@ def get_locations
+          resource_name = resource.fetch_path('properties', 'targetResource', 'resourceName')
+          uid = resource_uid(@subscription_id, group, resource_type.downcase, resource_name)
+
+          @resource_to_stack[uid] = stack_uid
+          child_stacks << uid if resource_type.downcase == TYPE_DEPLOYMENT
+          @data_index.fetch_path(:orchestration_stack_resources, uid)
+        end
+
+        return child_stacks, resources
+      end
+
+      def parse_stack_template(deployment, content)
+        # Only need a temporary unique identifier for the template. Using the stack id is the cheapest way.
+        uid = deployment.fetch_path('id')
+        ver = deployment.fetch_path('properties', 'templateLink', 'contentVersion')
+
+        new_result = {
+          :type        => "OrchestrationTemplateAzure",
+          :name        => deployment.fetch_path('name'),
+          :description => "contentVersion: #{ver}",
+          :content     => content
+        }
+        return uid, new_result
+      end
+
+      def parse_stack_parameter(param_key, param_obj, stack_id)
+        uid = resource_uid(stack_id, param_key)
+        new_result = {
+          :ems_ref => uid,
+          :name    => param_key,
+          :value   => param_obj['value']
+        }
+        return uid, new_result
+      end
+
+      def parse_stack_output(output_key, output_obj, stack_id)
+        uid = resource_uid(stack_id, output_key)
+        new_result = {
+          :ems_ref     => uid,
+          :key         => output_key,
+          :value       => output_obj['value'],
+          :description => output_key
+        }
+        return uid, new_result
+      end
+
+      def parse_stack_resource(resource, group)
+        status_message = resource.fetch_path('properties', 'statusMessage')
+        new_result = {
+          :ems_ref                => resource.fetch_path('properties', 'targetResource', 'id'),
+          :name                   => resource.fetch_path('properties', 'targetResource', 'resourceName'),
+          :logical_resource       => resource.fetch_path('properties', 'targetResource', 'resourceName'),
+          :physical_resource      => resource.fetch_path('properties', 'trackingId'),
+          :resource_category      => resource.fetch_path('properties', 'targetResource', 'resourceType'),
+          :resource_status        => resource.fetch_path('properties', 'provisioningState'),
+          :resource_status_reason => status_message || resource.fetch_path('properties', 'statusCode'),
+          :last_updated           => resource.fetch_path('properties', 'timestamp')
+        }
+        uid = resource_uid(@subscription_id, group, new_result[:resource_category].downcase, new_result[:name])
+        return uid, new_result
+      end
+
+      # Compose an id string combining some existing keys
+      def resource_uid(*keys)
+        keys.join('\\')
+      end
```

Comment:
In amzon `refresh_parser` we call it `compose_ems_ref`. I renamed it because the main purpose is really to compose an `uid` for `@data`. Is `compose_resource_uid` a better name? This is a private method though.